src/simtools/simtel/simtel_io_event_writer.py

```diff
@@ -15,144 +18,265 @@
+        """Process input files and return tables."""
+        # Process all files
+        for i, file in enumerate(self.input_files[: self.max_files]):
+            self._logger.info(f"Processing file {i + 1}/{self.max_files}: {file}")
+            self._process_file(i, file)
+
+        return self._create_tables()
+
+    def _create_tables(self):
+        """Create astropy tables from collected data."""
+        tables = []
+        for data, schema, name in [
+            (self.shower_data, TableSchemas.shower_schema, "SHOWERS"),
+            (self.trigger_data, TableSchemas.trigger_schema, "TRIGGERS"),
+            (self.file_info, TableSchemas.file_info_schema, "FILE_INFO"),
+        ]:
+            table = Table(rows=data, names=schema.keys())
+            table.meta["EXTNAME"] = name
+            self._add_units_to_table(table, schema)
+            tables.append(table)
+        return tables
+
+    def _add_units_to_table(self, table, schema):
+        """Add units to a single table's columns."""
+        for col, (_, unit) in schema.items():
+            if unit is not None:
+                table[col].unit = unit
+
+    def write(self, output_file, tables):
         """
-        Return shower and triggered event data.
-
-        Returns
-        -------
-        ShowerEventData, TriggeredEventData
-            Shower and triggered event data.
+        Write tables to file.
+
+        Parameters
+        ----------
+        output_file : Path
+            Path to the output file.
+        tables : list
+            List of astropy tables to write.
         """
-        return self.event_data, self.triggered_data
+        self._logger.info(f"Save reduced dataset to: {output_file}")
+        if output_file.name.endswith("fits.gz") or output_file.suffix == ".fits":
+            self._write_fits(tables, output_file)
+        elif output_file.suffix in (".h5", ".hdf5"):
+            self._write_hdf5(tables, output_file)
+        else:
+            raise ValueError(
+                f"Unsupported file format: {output_file.suffix}. "
+                "Supported formats are .fits and .hdf5"
+            )
+
+    def _write_fits(self, tables, output_file):
+        """Write tables to a FITS file."""
+        hdu_list = [fits.PrimaryHDU()]  # Primary HDU is required
+
+        for table in tables:
+            hdu = fits.table_to_hdu(table)
+            hdu.name = table.meta.get("EXTNAME", "")  # Set extension name if present
+            hdu_list.append(hdu)
+
+        fits.HDUList(hdu_list).writeto(output_file, overwrite=True)
+
+    def _write_hdf5(self, astropy_tables, output_file):
+        """Write tables to an HDF5 file."""
+        if importlib.util.find_spec("h5py") is None:
+            raise ImportError("h5py is required to write HDF5 files with Astropy.")
+
+        astropy_tables[0].write(
+            output_file,
+            path=f"/{astropy_tables[0].meta['EXTNAME']}",
+            format="hdf5",
+            overwrite=True,
```

Comment:
Do we want to overwrite by default? If this takes a while maybe better to have a error/prompt here?

Reply:
Added an overwrite option to the function option. On a general level, we should solve this question for all tools: see new issue #1585

Reply:
Yes I agree this would be important for some applications